src/js/common/schemaform/SaveStatus.jsx

```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import moment from 'moment';
+
+import { SAVE_STATUSES } from './save-load-actions';
+
+class SaveStatus extends React.Component {
+  render() {
+    const { form, profile } = this.props;
+
+    let savedAt;
+    let savedAtMessage;
+    const savedForm = profile && profile.savedForms.length > 0 && profile.savedForms
+      .filter(f => moment.unix(f.metadata.expires_at).isAfter())
+      .find(f => f.form === form.formId);
+    if (savedForm || form.lastSavedDate) {
+
+      savedAt = form.lastSavedDate
```

Comment:
I think we set `lastSavedDate` whenever we've auto-saved? You also have an `autoSuccess` status. I think you could simplify this logic and just use those values.

Reply:
That's much cleaner, change made.